### ISSUE TITLE
Intermittent incorrect navigation to and from (sub)views

### ISSUE THREAD

**Issue by Mariusthvdb** [home-assistant/frontend#19628]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.
- [ ] I have tried reproducing the issue in [safe mode](https://www.home-assistant.io/blog/2023/11/01/release-202311/#restarting-into-safe-mode) to rule out problems with unsupported custom resources.

### Describe the issue you are experiencing

since beta 2024.2 navigating to and from (sub)views is acting up. If it does, it always takes to the first view in the Dashboard, or when subviews are concerned, to the first subview in my yaml listing.

### Describe the behavior you expected

navigating like before, has been flawless since introduction

### Steps to reproduce the issue

1. click a button
2.
3.
...

### What version of Home Assistant Core has the issue?

2024.2.0bx

### What was the last working version of Home Assistant Core?

2024.1

### In which browser are you experiencing the issue with?

any

### Which operating system are you using to run this browser?

happening on desktop and Mobile

### Problem-relevant frontend configuration

```yaml
tap_action:
    action: navigate
    navigation_path: >
      [[[ return variables.path; ]]]
```

on a `custom:button-card` (which I checked btw for changes, and the dev confirmed nothing changed in that regard)

and a frontend config like:

to a subview:

```
      - type: custom:button-card
        template: subview
        icon: mdi:home-lightbulb
        variables:
          tooltip: Alle lampen
          path: /ui-sub-views/alle_lampen
```

to a regular view:

```
      - type: custom:button-card
        template: subview
        icon: mdi:power-socket-eu
        variables:
          tooltip: Schakelaars
          path: /ui-overzicht/schakelaars
```

### Javascript errors shown in your browser console/inspector

no errors at all

### Additional information

of course the first thing I did was changing the cards  to a core button and fixed path:
```
      - type: button
        name: Hue scenes
        icon: mdi:lightbulb
        tap_action:
          action: navigate
          navigation_path:  /ui-sub-views/hue_scenes

      - type: button
        name: Alle lampen
        icon: mdi:home-lightbulb
        tap_action:
          action: navigate
          navigation_path: /ui-sub-views/alle_lampen
```

but they too fluke regularly

when navigating to a subview is going wrong, the browser always lands on the first subview in my subviews dashboard yaml config:
```
views: 

  - !include ui-sub-views/subview_energieprijzen.yaml
  - !include ui-sub-views/subview_hue_scenes.yaml
  - !include ui-sub-views/subview_alle_lampen.yaml
  - !include ui-sub-views/subview_computer_netwerk.yaml
  - etc

  ```

if a regular view is in order and navigating goes wrong, the browser is taken to the first view in that particular Dashboard.

During all of those mistakes, the address in the browser is correct, and showing the correct target.

**Comment by karwosts** [comment]
Have you tried reproducing with a minimal example? I did a quick test but don't see anything unusual on my end. 

![navigation](https://github.com/home-assistant/frontend/assets/32912880/2ab41678-7cc9-4bb0-9361-44a57640263a)

**Comment by Mariusthvdb** [comment]
well, as always, 1 single instance in a minimal example is not a real life situation, nor would it probably trigger the issue.

not sure it would help in this case, but yes, I do have views with only a single navigation to a (sub)view.
and yes I tested that, and the issue remains.

In testing, I have just re-ordered my list of subviews to be alphabetical, and lo and behold, now all wrong navigations go to 
```
  - !include ui-sub-views/subview_alle_lampen.yaml
```

what I also noticed now (again, not 100% repeatable but still), is the tab indicator not completely moving from 1 tab to the other, but underlining the complete distance between 2 tabs. Somehow it seems the dashboard re-rendering is not complete, or gets stuck.

Ill ry and catch a screenshot of that
